pandas/core/frame.py

```diff
@@ -374,8 +376,12 @@ def _constructor_expanddim(self):
     # ----------------------------------------------------------------------
     # Constructors
 
-    def __init__(self, data=None, index=None, columns=None, dtype=None,
-                 copy=False):
+    def __init__(self,
+                 data=None,
+                 index: ArrayLike = None,
+                 columns: ArrayLike = None,
+                 dtype: Dtype = None,
```

Comment:
Optional[Dtype]

Reply:
Yeah I totally forgot about optional

Reply:
If the default argument is None you don’t need optional - Mypy adds for you

Reply:
@WillAyd didn't know that, but I actually like Optional as its very indicative

Reply:
I am going with Optional, being a bit explicit doesn't hurts, even though it's pretty clear the parameters are optional even without writing `Optional`.